_posts/2018-03-23-minnowboard-turbot-remote-firmware-flashing-with-remote-testing-environment.md

```diff
@@ -0,0 +1,312 @@
+---
+post_title: Minnowboard Turbot  remote firmware flashing with RTE (Remote Testing Environment)
+author: Arkadiusz Cichocki
+post_excerpt: ""
+layout: post
+published: true
+post_date: 2018-03-23 12:00:00
+tags:
+  - RTE
+  - Minnowboard Turbot
+  - Robot Framework
+  - firmware
+  - ROM
+  - remote development
+  - remote testing
+categories:
+
+---
+
+# Minnowboard Turbot  remote firmware flashing with RTE (Remote Testing Environment)
+
+
+Introduction
+------------
+
+Work related to a hardware carries some restrictions which don't occur when
+working only with a software. One of them is limited number of devices.
+This one may cause a problem with a accessibility to the platform. Limited 
+number of users could slow development and testing.
+What is more work with a harware requires a minimal knowledge about theory 
+of circuits and signals to elimite platform damage by an user. 
+Hardware can be expensive too.
+
+Remote Testing Environment project was made to resolve mentioned problems.
+The result of work is shown below:
+![](https://3mdeb.com/wp-content/uploads/2017/07/rte1-e1521807540813.jpg)
+![](https://3mdeb.com/wp-content/uploads/2017/07/rte2-e1521807599613.jpg)
+
+That platform makes possible to work on a hardware remotely.
+RTE connected to the tested device can provide some more advantages like:
+* Possibility to built testing system only once and place it in separated
+and secure place.
+* Advantage described above allows to keep order and eliminate time wasting
+for items search and connect.
+* Reduced number of hardware requirements for the user. Now he needs only a
+computer with a net connection and installed tools.
+
+Selected RTE functionality:
+* SPI with a header,
+* I2C with a header,
+* RS232,
+* OC buffer outputs header,
+* GPIO,
+* USB,
+* built-in relay with DC Jack 2.5/5.5 mm connectors to DUT power supply control.
```

Comment:
@acihy please add information that RTE is open source hardware project and please point to published repositories with schematics.

Reply:
@pietrushnic Repository with RTE schematics is located only on Gitlab at this moment.
Can I create a new 3mdeb repository on Github on my own and copy the RTE files to the new one?
I'm not sure that I have all required rights.

Reply:
@acihy https://github.com/3mdeb/rte-schematics whole 3mdeb team have permissions.